File: Week 1/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    // get input from the user: the amount of change
    float n;
    do
    {
        n = get_float("Change owed: ");
    }
    while (n < 0);

    // convert dollars to cents
    float c = n * 100;
    c = round(c);
    int d = c;

    // new variable to count the amount of coins
    int e = e;

    // subtract 25 cents from the amount as many times as possible
    while (d >= 25)
    {
        (d = d - 25);
        (e = e + 1);
    }

    // subtract 10 cents from the remaining amount as many times as possible
    while (d >= 10)
    {
        (d = d - 10);
        (e = e + 1);

    }

    // subtract 5 cents from the remaining amount as many times as possible
    while (d >= 5)
    {
        (d = d - 5);
        (e = e + 1);
    }

    // subtract 1 cent from the remaining amount as many times as possible
    while (d >= 1)
    {
        (d = d - 1);
        (e = e + 1);
    }

    // Print e: the amount of coins used
    {
        printf("%i\n", e);
    }

}

File: Week 6/cash.py
# Problem Set 6
# Name: Maria Daan
# Student Number: 11243406
#
# This program calculates the minimum number
# of coins required to give a user change.

from cs50 import get_float

# check for valid input
while True:
    n = get_float("Change owed: ")
    if n > 0:
        break

# convert to cents and define cointypes
cents = round(n * 100)
types = (25, 10, 5, 1)
coins = 0

# count how many coins are used
for i in range(len(types)):
    coins += cents // types[i]
    cents = cents % types[i]

print(f"{coins}")
